### PR TITLE
Market Maker Page

### PR DESCRIPTION
**Pull request by johnrjj**
Ready

### CODE REVIEW

File: ts/pages/staking/staking_pool.tsx [hunk 1]
```diff
@@ -240,43 +246,58 @@ const tradingPairs = [
...
-                isVerified={true}
-                estimatedStake={75}
-                rewardsShared={74}
-                iconUrl=""
+                title={stakingPool.metaData.name}
+                websiteUrl={stakingPool.metaData.websiteUrl}
+                poolId={utils.getAddressBeginAndEnd(stakingPool.operatorAddress)}
+                isVerified={stakingPool.metaData.isVerified}
+                estimatedStake={75} // what's this
+                rewardsShared={74} // what's this
```
Comment (johnrjj) [inline ts/pages/staking/staking_pool.tsx]: what are these two values?

File: ts/pages/staking/staking_pool.tsx [hunk 2]
```diff
@@ -240,43 +246,58 @@ const tradingPairs = [
...
+                iconUrl={stakingPool.metaData.logoUrl || ''}
                 tabs={[
                     {
                         title: 'Current Epoch',
                         metrics: [
                             {
                                 title: 'Total Volume',
-                                number: '1.23M USD',
+                                number: '1.23M USD', // volume missing
```
Comment (johnrjj) [inline ts/pages/staking/staking_pool.tsx]: can this be added to the api or do we want to skip this

File: ts/pages/staking/staking_pool.tsx [hunk 3]
```diff
@@ -240,43 +246,58 @@ const tradingPairs = [
...
                             },
                             {
                                 title: 'ZRX Staked',
-                                number: '1,288,229',
+                                number: '1,288,229', // zrx staked missing
```
Comment (johnrjj) [inline ts/pages/staking/staking_pool.tsx]: total stake is missing from the poolById response

File: ts/pages/staking/staking_pool.tsx [hunk 4]
```diff
@@ -337,16 +359,18 @@ export const StakingPool: React.FC<StakingPoolProps & RouteChildrenProps> = prop
...
                 <GraphHeading>Historical Details</GraphHeading>
                 <HistoryChart
-                    fees={[40, 41, 40, 41, 40, 41, 40, 41, 40, 41, 40, 41]}
-                    rewards={[30, 31, 30, 31, 30, 31, 30, 31, 30, 31, 30, 31]}
-                    epochs={[3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]}
+                    fees={stakingPool.epochRewards.map(e => e.totalRewardsPaidInEth)}
+                    rewards={stakingPool.epochRewards.map(e => e.totalRewardsPaidInEth)}
+                    epochs={stakingPool.epochRewards.map(() => 3)} // magic number
+                    // timestamps are missing (stakingPool.epochRewards.timestamp or something)
                     labels={['1 July', '5 July', '10 July', '15 July', '20 July', '25 July', '30 July']}
```
Comment (johnrjj) [inline ts/pages/staking/staking_pool.tsx]: need timestamps for epochs

File: ts/pages/staking/staking_pool.tsx (hunk 1 above)
Comment (johnrjj) [reply ts/pages/staking/staking_pool.tsx]: rewards shared I'm guessing is just user rewards / total rewards, but not sure what `estimatedStake` is

**Comment by johnrjj** [comment]
Depends on https://github.com/0xProject/0x-api/pull/45

File: ts/components/staking/dashboard_hero.tsx [hunk 5]
```diff
@@ -234,7 +236,15 @@ export const DashboardHero: React.FC<DashboardHeroProps> = ({
             <Inner>
                 <Row>
                     <Column>
-                        <PoolIcon src={iconUrl} />
+                    {
+                        iconUrl
+                            ? <PoolIcon src={iconUrl} />
+                            : <Jazzicon
+                                seed={1}
```
Comment (kimpers) [inline ts/components/staking/dashboard_hero.tsx]: Should `seed` not be poolId?

File: ts/pages/staking/staking_pool.tsx [hunk 6]
```diff
@@ -240,70 +248,85 @@ const tradingPairs = [
...
+    const networkId = useSelector((state: State) => state.networkId);
+    const apiClient = useAPIClient(networkId);
+    const [stakingPool, setStakingPool] = useState<PoolWithHistoricalStats | undefined>(undefined);
+
+    useEffect(() => {
+        apiClient
+            .getStakingPoolByIdAsync(poolId)
+            .then(res => setStakingPool(res.stakingPool))
+            .catch(e => logUtils.warn(e));
+    }, [poolId, setStakingPool, apiClient]);
```
Comment (kimpers) [inline ts/pages/staking/staking_pool.tsx:260]: Wouldn't `apiClient` have a new reference on each render given that we just instantiate and return a new instance in the hook? Or is this magically handled by React?

```suggestion
    }, [poolId, setStakingPool, apiClient.account]);
```

**Review by kimpers — APPROVED** [review]

File: ts/pages/staking/staking_pool.tsx (hunk 6 above)
Comment (johnrjj) [reply ts/pages/staking/staking_pool.tsx:260]: no the useApiClient hook holds on to the same reference of apiClient unless the network Id changes

File: ts/components/staking/dashboard_hero.tsx (hunk 5 above)
Comment (johnrjj) [reply ts/components/staking/dashboard_hero.tsx]: good catch! fixed ,thanks!

File: ts/pages/staking/staking_pool.tsx (hunk 6 above)
Comment (johnrjj) [reply ts/pages/staking/staking_pool.tsx:260]: (if you're curious

https://github.com/0xProject/website/blob/development-staking/ts/hooks/use_api_client.ts#L6
this ref holds on to the reference between renders

and will only update when networkId changes -- https://github.com/0xProject/website/blob/development-staking/ts/hooks/use_api_client.ts#L8-L10
)

Comment (johnrjj) [reply ts/pages/staking/staking_pool.tsx:260]: (I had refactored it a little bit ago to be more 'hook'ey)

**Pull request merged**